### PR TITLE
feat: add outlineColor prop in TextInputOutlined & add titleStyle prop in MenuItem & add wrapInSafeArea prop in Modal

### PR DESCRIPTION
**Pull request by ahce**
#### Modal
Determines whether the modal is wrapped in a SafeAreaView
#### MenuItem
Allows change title style.
#### TextInput
Allows to controls the outline color.
Example with outlineColor='transparent'

#### Normal input
image
#### Focused input
image

### Test plan

1. Download this branch
2. Open the TextInput or MenuItem or Modal example
3. Change the prop outlineColor (TextInput) or titleStyle (MenuItem) or wrapInSafeArea (Modal)

### CODE REVIEW

File: src/components/Modal.tsx [hunk 1]
```diff
@@ -178,8 +183,15 @@ class Modal extends React.Component<Props, State> {
...
-    const { children, dismissable, theme, contentContainerStyle } = this.props;
+    const {
+      children,
+      dismissable,
+      wrapInSafeArea = true,
+      theme,
+      contentContainerStyle,
+    } = this.props;
     const { colors } = theme;
+    const Wrapper = wrapInSafeArea ? SafeAreaView : View;
```
Comment (Trancever) [inline src/components/Modal.tsx]: Why do you want to remove SafeAreaView? Do you have any issues with it?

File: src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -55,6 +55,10 @@ export type TextInputProps = React.ComponentPropsWithRef<
    * Underline color of the input.
    */
   underlineColor?: string;
+  /**
+   * Outline color of the input.
+   */
+  outlineColor?: string;
```
Comment (Trancever) [inline src/components/TextInput/TextInput.tsx]: We need to rethink the strategy of customizing a TextInput. We won't accept this change now. Please give us some more time and we will come up with a good plan for both Flat and Outlined TextInput.

File: src/components/Modal.tsx (hunk 1 above)
Comment (ahce) [reply src/components/Modal.tsx]: Yeah problems with the background color, also when I want to stretch an image in the whole modal

File: src/components/TextInput/TextInput.tsx (hunk 2 above)
Comment (ahce) [reply src/components/TextInput/TextInput.tsx]: I don't see high impact on this change.
Is there any chance that you will approve it or should I reverse it?

Comment (Trancever) [reply src/components/TextInput/TextInput.tsx]: It should be done in a separate PR anyway, so yes, please revert this change.

File: src/components/Modal.tsx (hunk 1 above)
Comment (Trancever) [reply src/components/Modal.tsx]: Could you provide a [Snack](https://snack.expo.io/) that reproduces your issue? Thanks in advance!

**Pull request closed without merging**